### PR TITLE
Add build instructions

### PR DESCRIPTION
**Pull request by chsasank**
As promised in #335, here are the instructions for building this fork of triton. Can add a link to this doc in README if maintainers are ok :)

### CODE REVIEW

File: docs/build.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+## Dependencies
+
+### Drivers
+
+To build triton from source, we need to start with installing GPU drivers. If you're using Data Center (PVC) GPUs, please follow the instructions from [here](https://dgpu-docs.intel.com/driver/installation.html#install-steps) of your Linux distribution. For Intel Arc (client) GPUs, here are the [instructions](https://dgpu-docs.intel.com/driver/client/overview.html).
+
+
+### Python and Virtual Environment
+
+Next, we install python3 and venv in our system. In ubuntu, follow these instructions
```
Comment (pbchekin) [inline docs/build.md:12]:
```suggestion
Next, we install python3 and venv in our system. For Ubuntu 22.04, follow these instructions
```

File: docs/build.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+## Build from Source
+
+First clone this repository:
+
+```
+git clone https://github.com/intel/intel-xpu-backend-for-triton.git -b llvm-target
+cd intel-xpu-backend-for-triton
+```
+
+Set up dpc++ environment variables. This works on Ubuntu, you might have to change the directories based on installation.
```
Comment (pbchekin) [inline docs/build.md:31]: This requires oneAPI BaseKit mentioned here: https://github.com/intel/intel-xpu-backend-for-triton/blob/llvm-target/README.md?plain=1#L26.

File: docs/build.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+### Python and Virtual Environment
+
+Next, we install python3 and venv in our system. In ubuntu, follow these instructions
+
+```
+sudo apt update
+sudo apt install -y python3 python3-pip python3-venv
+```
+
+In other Linux distributions, you might have to follow different instructions.
```
Comment (pbchekin) [inline docs/build.md:19]: It would be nice to mention that Python needs to be 3.9 or 3.10 for other distributions.

**Review by pbchekin — COMMENTED** [review]
Thank you for the contribution, @chsasank!

File: docs/build.md (hunk 3 above)
Comment (chsasank) [reply docs/build.md]: done

File: docs/build.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
+# Building Triton for Intel GPUs
+
+## Dependencies
+
+### Drivers and OneAPI Toolkit
+
+To build triton from source, we need to start with installing GPU drivers. If you're using Data Center (PVC) GPUs, please follow the instructions from [here](https://dgpu-docs.intel.com/driver/installation.html#install-steps) of your Linux distribution. For Intel Arc (client) GPUs, here are the [instructions](https://dgpu-docs.intel.com/driver/client/overview.html).
+
+We also need to install OneAPI Basekit with version > 2024.0.1. Please follow the instructions from [here](https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit-download.html) to install it.
```
Comment (pbchekin) [inline docs/build.md:9]:
```suggestion
We also need to install OneAPI Basekit with version >= 2024.0.1. Please follow the instructions from [here](https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit-download.html) to install it.
```

File: docs/build.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+```
+$ source $BASE/intel-xpu-backend-for-triton/.venv/bin/activate
+$ python
+Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
+Type "help", "copyright", "credits" or "license" for more information.
+>>> import intel_extension_for_pytorch
+>>> import triton
+```
+
+This should not throw any error!
```
Comment (pbchekin) [inline docs/build.md:63]: Please add trailing new line to pass pre-commit check:
```suggestion
This should not throw any error!

```

**Review by pbchekin — APPROVED** [review]
Looks good!

**Comment by jinz2014** [comment]
Thank you for adding the instruction.  There are some comments:

source ${ONEAPIROOT}/setvars.sh  may be the only command needed to set up the oneAPI environment before building Triton. 

The installed version is triton-3.0.0.

**Comment by chsasank** [comment]
Do you want me to create another PR editing these?

**Pull request merged**